assets/src/components/mapPage/searchResultsByProperty.tsx

```diff
@@ -116,83 +82,110 @@ const LocationSearchResultSection = ({
       >
         {searchPropertyDisplayConfig.location.name}
       </h2>
-      {locationSearchResults === null ? (
+      {isLoading(results) ? (
         <Loading />
-      ) : locationSearchResults.length > 0 ? (
+      ) : (
         <>
           <ul className="c-search-results__list">
-            {locationSearchResults
-              .slice(0, limit)
-              .map((locationSearchResult) => (
-                <li key={locationSearchResult.id}>
-                  <Card
-                    style="white"
-                    title={
-                      locationSearchResult.name || locationSearchResult.address
-                    }
-                    openCallback={() => onSelectLocation(locationSearchResult)}
-                  >
-                    {locationSearchResult.name &&
-                      locationSearchResult.address && (
-                        <CardBody>{locationSearchResult.address}</CardBody>
-                      )}
-                  </Card>
-                </li>
-              ))}
+            {results.ok.matches.map((locationSearchResult) => (
+              <li key={locationSearchResult.id}>
+                <Card
+                  style="white"
+                  title={
+                    locationSearchResult.name || locationSearchResult.address
+                  }
+                  openCallback={() => onSelectLocation(locationSearchResult)}
+                >
+                  {locationSearchResult.name &&
+                    locationSearchResult.address && (
+                      <CardBody>{locationSearchResult.address}</CardBody>
+                    )}
+                </Card>
+              </li>
+            ))}
           </ul>
-          {locationSearchResults.length > limit && (
-            <div className="c-map_page__search_results_actions">
-              <button
-                className="c-map-page__show_more button-text"
-                onClick={() => showMore()}
-              >
-                Show more
-              </button>
-            </div>
-          )}
+          {results.ok.hasMoreMatches && <ShowMore property={"location"} />}
         </>
-      ) : (
-        "No results found"
       )}
     </section>
   )
 }
 
+const ShowMore = ({ property }: { property: SearchProperty }) => {
```

Comment:
**praise:** thanks for pulling this out into a separate component, I should have done this originally in my PR.